examples/fit-model-ceres.cpp

```diff
@@ -18,61 +18,240 @@
  * limitations under the License.
  */
 #define GLM_FORCE_UNRESTRICTED_GENTYPE
-#include "eos/core/Landmark.hpp"
-#include "eos/core/LandmarkMapper.hpp"
-#include "eos/core/read_pts_landmarks.hpp"
-#include "eos/fitting/ceres_nonlinear.hpp"
-#include "eos/fitting/contour_correspondence.hpp"
-#include "eos/fitting/fitting.hpp"
-#include "eos/morphablemodel/Blendshape.hpp"
 
-#include "Eigen/Core"
-
-#include "glm/ext.hpp"
-#include "glm/glm.hpp"
-
-#include "ceres/ceres.h"
-#include "ceres/cubic_interpolation.h"
-#include "ceres/rotation.h"
+#include <chrono>
+#include <iostream>
+#include <stdexcept>
+#include <vector>
 
-#include "opencv2/core/core.hpp"
+#include "boost/filesystem.hpp"
+#include "boost/program_options.hpp"
 #include "opencv2/imgproc/imgproc.hpp"
 #include "opencv2/highgui/highgui.hpp"
 
-#include "boost/filesystem.hpp"
-#include "boost/program_options.hpp"
+#include "eos/core/Landmark.hpp"
+#include "eos/core/LandmarkMapper.hpp"
+#include "eos/core/read_pts_landmarks.hpp"
+#include "eos/fitting/ceres_nonlinear.hpp"
+#include "eos/morphablemodel/Blendshape.hpp"
 
-#include <chrono>
-#include <iostream>
-#include <vector>
 
 using namespace eos;
-using namespace glm;
 using namespace ceres;
 namespace po = boost::program_options;
 namespace fs = boost::filesystem;
 using eos::core::Landmark;
 using eos::core::LandmarkCollection;
+using eos::core::IndexedLandmark;
+using eos::core::IndexedLandmarkCollection;
 using cv::Mat;
 using Eigen::Vector2f;
-using std::cout;
-using std::endl;
-using std::string;
-using std::vector;
-
-// print a vector:
-template <typename T>
-std::ostream& operator<<(std::ostream& out, const std::vector<T>& v)
-{
-    if (!v.empty())
-    {
+
+
+#ifndef EOS_CERES_EXAMPLE_USE_PERSPECTIVE
+#define EOS_CERES_EXAMPLE_USE_PERSPECTIVE true
```

Comment:
Is it possible to put these defines as variables in `main` instead? Perhaps as `constexpr`, so they would be compile-time too? (It should be supported by C++14 I think?)

Reply:
The idea was to have an option to pass these arguments as compile `-D` parameters. What do you think about it?